atoma-proxy/src/main.rs

```diff
@@ -104,21 +114,38 @@ async fn main() {
     let args = Args::parse();
     let config = Config::load(args.config_path).await;
 
-    let (shutdown_sender, shutdown_receiver) = watch::channel(false);
+    let (_shutdown_sender, shutdown_receiver) = watch::channel(false);
     let (event_subscriber_sender, event_subscriber_receiver) = flume::unbounded();
     let (state_manager_sender, state_manager_receiver) = flume::unbounded();
 
-    let sui_subscriber =
-        atoma_sui::SuiEventSubscriber::new(config.sui, event_subscriber_sender, shutdown_receiver);
+    let sui_subscriber = atoma_sui::SuiEventSubscriber::new(
+        config.sui.clone(),
+        event_subscriber_sender,
+        shutdown_receiver.clone(),
+    );
 
     // Initialize your StateManager here
     let state_manager = AtomaStateManager::new_from_url(
         config.state.database_url,
         event_subscriber_receiver,
         state_manager_receiver,
     )
-    .await;
-    if let Err(e) = state_manager {
-        error!("State manager error {}", e);
-    }
+    .await
+    .unwrap();
```

Comment:
Do not unwrap, I understand that in this situation it might be best to panic on state manager initialization, but in that case, I recommend using `.expect` with a proper panicking message.